kore.maude

```diff
@@ -220,18 +227,30 @@ Questions about this definition:
 - do we really need ~> as a K construct?  Can it be a label and thus the way K
   deal with evaluation contexts just a particular methodology and not an
   intrisic part of K ? Same question for =>.
+  Cosmin: Syntacticly at least, I would say "no" for both. From the implementation
+  perspective, this will simplify traversal code even further. 
 - rename .::K and .::KList to . ?  if we get rid of ~> as above, then we may
   only need .::KList.
+  Cosmin: I agree.
 - Should we allow empty productions instead of sort declarations?  A pro
   argument is that we have fewer syntactic constructs.  A cons argument is
   that a sort declaration may be regarded as a different entityr, with
   different kinds of attributes.
+  Cosmin: I don't quite follow.
 - Should we add constant AttributeKeys which have special semantics, like
   "strict", as constants?
+  Cosmin: we may want to discuss the attribute syntax separately.
 - Brackets and disappearing injections are the same thing, arent's they?  A
   production which only has one non-terminal and the production disappears
   from the AST.
+  Cosmin: Yes, I also think they can be seen as the same thing. That being
+          said, I would really like for injections to not dissapear. If something
+          is meant to dissapear, it should be declared as a subsort. And the
+          dissapearing injections (including subsorts) could be done via rules as
+          we discussed before.
```

Comment:
I understand this and agree. The problem is that we do have sorts now, and many passes in the compiler and the backend rely on them. Removing them is not something we can solve in the short term. As we don't want them there eventually, we can avoid adding the subsort sentence to the KORE syntax. But I do want it there for the implementation. This will actually help when eventually removing sorts, as we have the productions clearly identified (via Java typing, not just checking for an elusive attribute).

So, my concrete proposal is:
1. Have `syntax A ::= B` actually mean an injection, not a subsort. Subsorts would need to be `syntax A ::= B [subsort]`
2. The data structures will have separate `Production` and `Subsort` classes, instantiated depending on the subsort attribute.
3. Eliminate `syntax A` and replace it with `syntax K ::= A [subsort]` (I don't feel strongly about this)

What do you think?